src/pint/models/noise_model.py

```diff
@@ -606,6 +610,179 @@ def pl_dm_cov_matrix(self, toas: TOAs) -> np.ndarray:
+                description="Number of SW DM noise frequencies.",
+                convert_tcb2tdb=False,
+            )
+        )
+        self.add_param(
+            floatParameter(
+                name="TNSWFLOG",
+                units="",
+                description="Number of logarithmically solar wind frequencies in the basis.",
+                convert_tcb2tdb=False,
+            )
+        )
+        self.add_param(
+            floatParameter(
+                name="TNSWFLOG_FACTOR",
+                units="",
+                description="Factor of the log-spaced solar wind frequencies (2 -> [1/8,1/4,1/2,...])",
+                convert_tcb2tdb=False,
+            )
+        )
+
+        self.covariance_matrix_funcs += [self.pl_sw_cov_matrix]
+        self.basis_funcs += [self.pl_sw_basis_weight_pair]
+
+    def get_plc_vals(self) -> Tuple[float, float, int, int, float]:
+        """
+        Retrieve power-law parameters and frequency-basis parameters
+        from the model, substituting defaults if unspecified.
+        """
+        n_lin = int(self.TNSWC.value) if self.TNSWC.value is not None else 100
+        n_log = int(self.TNSWFLOG.value) if (self.TNSWFLOG.value is not None) else None
+        sw_log_factor = (
+            self.TNSWFLOG_FACTOR.value
+            if (self.TNSWFLOG_FACTOR.value is not None)
+            else 2
+        )
+        amp, gam = 10**self.TNSWAMP.value, self.TNSWGAM.value
+        f_min_ratio = 1 / (sw_log_factor**n_log) if n_log is not None else 1
+
+        return amp, gam, n_lin, n_log, f_min_ratio
+
+    def get_time_frequencies(self, toas: TOAs) -> np.ndarray:
+        """Return the frequencies of the noise model"""
+
+        tbl = toas.table
+        t = (tbl["tdbld"].quantity * u.day).to(u.s).value
+        T = np.max(t) - np.min(t)
+
+        (_, _, n_lin, n_log, f_min_ratio) = self.get_plc_vals()
+        f_min = f_min_ratio / T
+
+        return t, get_rednoise_freqs(
+            t, n_lin, Tspan=T, logmode=0, f_min=f_min, nlog=n_log
+        )
+
+    def get_noise_basis(self, toas: TOAs) -> np.ndarray:
+        """Return a Fourier design matrix for SW DM noise.
+
+        See the documentation for pl_sw_basis_weight_pair function for details."""
+
+        freqs = self._parent.barycentric_radio_freq(toas).to(u.MHz)
+        # get the achromatic Fourier design matrix
+        t, f = self.get_time_frequencies(toas)
+        Fmat = create_fourier_design_matrix(t, f)
+        # get solar wind geometry from pint.models.solar_wind_dispersion.SolarWindDispersion
+        solar_wind_geometry = self._parent.solar_wind_geometry(toas)
+        # dispersion constant taken from Handbook of Pulsar Astronomy, eq. 4.5
+        # the cm**-3 at the end reflect the units of the GP perturbations which are solar electron number density
+        # since this is the SW DM value if n_earth = 1 cm^-3. the GP will scale it.
+        dt_DM = (
+            solar_wind_geometry
+            * 4.148808e3
```

Comment:
What is this constant?

Reply:
I meant use the already defined constant (`pint.DMconst`) instead of hardcoding the value.